xfields/beam_elements/beambeam3d.py

```diff
@@ -183,22 +222,40 @@ def __init__(self,
             self.iscollective = True
             self.track = self._track_collective # switch to specific track method
 
-            #assert slices_other_beam_zeta_center is not None
-            #assert not np.isscalar(slices_other_beam_num_particles)
-
-            if slices_other_beam_zeta_center is None: 
-                slices_other_beam_zeta_center = self.config_for_update.slicer.bin_centers
+            if slices_other_beam_zeta_center is None:
+                if isinstance(self.config_for_update.slicer, Slicer):
+                    slices_other_beam_zeta_center = self.config_for_update.slicer.bin_centers * self.config_for_update.slicer.sigma_z
+                else:
```

Comment:
Not very clear

Reply:
removed sigma_z. Now slicer takes in simga_z as a parameter and slices are scaled to it and are in units of [m].